### ISSUE TITLE
Custom loss value 8x of expected value on TPU v2-8. OK on TPU v5-1 and gpus

### ISSUE THREAD

**Issue by van-flow** [googlecolab/colabtools#5130]
**Describe the current behavior**
I am using a custom loss. When running on v5e-1 and gpus, loss behavior is as expected. When I switch to v2-8, loss is 8 times that, and often switches to nan after a few batches are processed with each epoch.

This was not the case earlier with the same notebook on TPU v2-8. Something has changed recently to create this odd behavior. But I am unable to pinpoint what that may be.

**Describe the expected behavior**
Expected behavior is 1x loss, but getting 8x or nan on TPU v2-8

**What web browser you are using**
Chrome

**Additional context**
tensorflow 2.18.0
tf.keras 3.8.0
python 3.11.11

**Comment by metrizable** [comment]
@van-flow Please provide a minimum reproducible notebook and re-open this issue.
